Library/Homebrew/utils/github.rb

```diff
@@ -613,25 +613,49 @@ def self.forked_repo_info!(tap_remote_repo, org: nil)
     [remote_url, username]
   end
 
-  def self.create_bump_pr(info, args:)
-    tap = info[:tap]
-    sourcefile_path = info[:sourcefile_path]
-    old_contents = info[:old_contents]
-    additional_files = info[:additional_files] || []
-    remote = info[:remote] || "origin"
-    remote_branch = info[:remote_branch] || tap.git_repo.origin_branch_name
-    branch = info[:branch_name]
-    commit_message = info[:commit_message]
-    previous_branch = info[:previous_branch] || "-"
-    tap_remote_repo = info[:tap_remote_repo] || tap.full_name
-    pr_message = info[:pr_message]
-
-    sourcefile_path.parent.cd do
-      git_dir = Utils.popen_read("git", "rev-parse", "--git-dir").chomp
-      shallow = !git_dir.empty? && File.exist?("#{git_dir}/shallow")
-      changed_files = [sourcefile_path]
-      changed_files += additional_files if additional_files.present?
+  # @param commits data on commits to apply consisting of the commit message,
+  #   the source file path, any additional files, and the old contents of file.
+  #   Must be non-empty and all source file paths must be part of same git repo.
+  sig {
+    params(
+      tap:             Tap,
+      branch:          String,
+      commits:         T::Array[[String, Pathname, T.nilable(T::Array[String]), String]],
```

Comment:
May switch to a `Struct` to be more explicit rather than having to read docs to know what index means.